docs/docs/manual/expressions.md

```diff
@@ -1,57 +1,414 @@
+
+#### GREL-supported regex
+
+To write a regular expression inside a GREL expression, wrap it between a pair of forward slashes (/) much like the way you would in Javascript. For example, in
+
+```
+value.replace(/\s+/, " ")
+```
+
+the regular expression is `\s+`, and the syntax used in the expression wraps it with forward slashes (`/\s+/`). Though the regular expression syntax in OpenRefine follows that of Java (normally in Java, you would write regex as a string and escape it like "\\s+"), a regular expression within a GREL expression is similar to Javascript.
+
+Do not use slashes to wrap regular expressions outside of a GREL expression.
+
+The [GREL functions](#grel-general-refine-expression-language) that support regex are:
+*   replace
+*   match
+*   partition
+*   rpartition
+*   split
+
+#### Jython-supported regex
+
+You can also use [regex with Jython expressions](http://www.jython.org/docs/library/re.html), instead of GREL, for example with a Custom Text Facet: 
+
+```
+python import re g = re.search(ur"\u2014 (.*),\s*BWV", value) return g.group(1)
+```
+
+#### Clojure-supported regex
+
+[Clojure](https://clojure.org/reference/reader) uses the same regex engine as Java, and can be invoked with [re-find](http://clojure.github.io/clojure/clojure.core-api.html#clojure.core/re-find), [re-matches](http://clojure.github.io/clojure/clojure.core-api.html#clojure.core/re-matches), etc. You can use the #"pattern" reader macro as described [in the Clojure documentation](https://clojure.org/reference/other_functions#regex). For example, to get the nth element of a returned sequence, you can use the nth function:
+
+```
+clojure (nth (re-find #"\u2014 (.*),\s*BWV" value) 1)
+```
+
+## Variables
+
+Most of the OpenRefine-specific variables have attributes: aspects of the variables that can be called separately. We call these attributes "member fields" because they belong to certain variables. For example, you can query a record to find out how many rows it contains with `row.record.rowCount`: `rowCount` is a member field specific to `record`, which is a member field of `row`. Member fields can be called using a dot separator, or with square brackets (`row["record"]`).
+
+|Variable |Meaning |
+|-|-|
+| value | The value of the cell in the current column of the current row (can be null) |
+| row | The current row |
+| row.record | One or more rows grouped together to form a record |
+| cells | The cells of the current row, with fields that correspond to the column names (or row.cells) |
+| cell | The cell in the current column of the current row, containing value and other attributes |
+| cell.recon | The cell's reconciliation information returned from a reconciliation service or provider |
+| rowIndex | The index value of the current row (the first row is 0) |
```

Comment:
I would format tables like this with inline code formatting in the first column: ``| `value` |`` instead of `| value |`. The same applies to all other tables which list variables, fields, or anything that would be typed in an expression editor.

Reply:
I'm torn on this one. With the latest update everything is now formatted this way, but I think I would prefer to have none of it (and remove what was there before). Tables don't really need formatting inside them, I think. 
But, it's done for now and can probably stay that way.